Adds daily spending receive guidance for BOLT 12 and splicing (#1194)
BitcoinDesign/Guide: guide/daily-spending-wallet/requesting/requesting.md

After entering an amount, the user taps the 'Request' button. The app then generates and presents a payment request.

This wallet defaults to a [unified payment request]({{ "/guide/how-it-works/payment-request-formats/#unified-payment-requests" | relative_url }}). This combines a [lightning invoice]({{ "/guide/how-it-works/payment-request-formats/#invoice" | relative_url }}), an [offer]({{ "/guide/how-it-works/payment-request-formats/#offers" | relative_url }}) and an [on-chain address]({{ "/guide/how-it-works/payment-request-formats/#addresses" | relative_url }}) into a single payment request using BIP(3)21.

Unified requests eliminate the hassle for users of choosing between requesting payment via the lightning network or on-chain. This choice can often be confusing, particularly for newcomers. 

> This wallet utilizes **splicing** for all on-chain transactions. Splicing allows the seamless transfer of received Bitcoin into the user's Lightning balance without maintaining a separate on-chain balance. As a result, the wallet presents a single Lightning balance, greatly simplifying the user experience. This approach is trust-minimized and non-custodial, as channel capacity can be increased or decreased directly on-chain without relying on external swap services. We cover these more in our [lightning liquidity]({{ "/guide/how-it-works/liquidity" | relative_url }}) and receiving pages.

Reviewer: This is existing language, but "**for** all on-chain transactions" tripped me up. I assume it only means receive-only?

Since added text mentions lightning liquidity and receiving pages, content will also have to be added/updated on those pages.

```suggestion
This wallet utilizes **splicing** for on-chain transactions. Splicing allows the seamless transfer of received Bitcoin into the user's Lightning balance without maintaining a separate on-chain balance. As a result, the wallet presents a single Lightning balance, greatly simplifying the user experience. This approach is trust-minimized and non-custodial, as channel capacity can be increased or decreased directly on-chain without relying on external swap services. We cover this in detail in our [lightning liquidity]({{ "/guide/how-it-works/liquidity" | relative_url }}) and receiving pages.
```

Reviewer 2: Hm, if the full balance is always on lightning, then a splice would also be needed for sending, no? To get the ₿ out of the channel?

Reviewer: Good question. This being the Requesting page, shouldn't the commentary be restricted to receiving anyways? On this front, this is a bit of a nit.

(disclaimer: lightning is not my wheelhouse, so I'd love to know what you and others who understand this better, think.)

Re: the meat of your question:

My understanding is that one of the main benefits of splicing for on-chain, is that it increases your channel size, which is strictly good, which also benefits other use cases. But using splicing for on-chain sends will reduce channel size, which is not desirable. For e.g. this might make subsequent lightning receives (use case 2 above) more likely, to require splicing with associated on-chain cost.

Perhaps swaps is an alternative approach that could be used (retained?) for on-chain sends. I know there are issues (costs, third party reliance?) with this too. I've asked Phoenix on twitter, let's see if they respond.

Reviewer: update: phoenix uses splice outs, but also – https://x.com/phoenixwallet/status/1970891555704799740

Reviewer 2: Reducing channel size can be desirable, no? When I have a big balance and want to move it to cold storage or so?

Reviewer: That's a good point, and a proper splice out use-case... (i wonder if there should be a proper flow for this to allow users to do this?)..,

### Payment link

> Payment links use a BIP(3)21 [URI]({{ "/guide/how-it-works/payment-request-formats/#uniform-resource-identifier-uris-schemes" | relative_url }}) `bitcoin:` which makes these readable by other bitcoin applications. These can be included as part of a button or hyperlink. Also see the [wallet selector UI pattern]({{ "/guide/how-it-works/wallet-selector/" | relative_url }}).

Reviewer: "BIP(3)21" tripped me up since I didn't know about BIP321, but it might be just me.

The linked page ([payment request formats](https://bitcoin.design/guide/how-it-works/payment-request-formats/#uniform-resource-identifier-uris-schemes)) only mentions BIP21, so this could be updated to mention and link to BIP321.

Reviewer 2: Probably fine to just write `BIP21` here. The `(3)` part reads funny, and BIP21 covers the relevant functionality.
